packages/calcite-components/src/components/swatch/swatch.tsx

```diff
@@ -0,0 +1,483 @@
+    Please refactor your code to reduce the need for this check.
+    Docs: https://qawebgis.esri.com/arcgis-components/?path=/docs/lumina-transition-from-stencil--docs#watching-for-property-changes */
+    if (changes.has("selected") && this.hasUpdated) {
+      this.watchSelected(this.selected);
+    }
+    if (changes.has("color")) {
+      this.handleColorChange(this.color);
+    }
+  }
+
+  override updated(): void {
+    updateHostInteraction(this);
+  }
+
+  loaded(): void {
+    if (this.selectionMode !== "none" && this.interactive && this.selected) {
+      this.handleSelectionPropertyChange(this.selected);
+    }
+  }
+
+  //#endregion
+
+  //#region Private Methods
+
+  private watchSelected(selected: boolean): void {
+    if (this.selectionMode === "none") {
+      return;
+    }
+    this.handleSelectionPropertyChange(selected);
+  }
+
+  private keyDownHandler(event: KeyboardEvent): void {
+    if (event.target === this.el) {
+      switch (event.key) {
+        case " ":
+        case "Enter":
+          this.handleEmittingEvent();
+          event.preventDefault();
+          break;
+        case "ArrowRight":
+        case "ArrowLeft":
+        case "Home":
+        case "End":
+          this.calciteInternalSwatchKeyEvent.emit(event);
+          event.preventDefault();
+          break;
+      }
+    }
+  }
+
+  private handleSlotImageChange(event: Event): void {
+    this.hasImage = slotChangeHasAssignedElement(event);
+  }
+
+  private handleEmittingEvent(): void {
+    if (this.interactive) {
+      this.calciteSwatchSelect.emit();
+    }
+  }
+
+  private handleSelectionPropertyChange(selected: boolean): void {
+    if (this.selectionMode === "single") {
+      this.calciteInternalSyncSelectedSwatches.emit();
+    }
+    const selectedInParent = this.parentSwatchGroup.selectedItems.includes(this.el);
+
+    if (!selectedInParent && selected && this.selectionMode !== "multiple") {
+      this.calciteInternalSwatchSelect.emit();
+    }
+    if (this.selectionMode !== "single") {
+      this.calciteInternalSyncSelectedSwatches.emit();
+    }
+  }
+
+  private handleColorChange(color: string | null): void {
+    this.internalColor = color ? Color(color) : null;
+  }
+
+  //#endregion
+
+  //#region Rendering
+
+  private renderSwatchImage(): JsxNode {
+    return (
+      <div class={CSS.imageContainer}>
+        <slot name={SLOTS.image} onSlotChange={this.handleSlotImageChange} />
+      </div>
+    );
+  }
+
+  private renderEmptyDisplay(): JsxNode {
+    const scale = this.scale === "s" ? "12" : this.scale === "m" ? "16" : "20";
+
+    return (
+      <div class={CSS.emptyContainer}>
+        <svg
+          fill="none"
+          height={scale}
+          viewBox={`0 0 ${scale} ${scale}`}
+          width={scale}
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <path d={`M${scale} 0L0 ${scale}`} stroke="#D83020" stroke-width="3" />
```

Comment:
Should this stroke be styled via CSS and available as a CSS var? Same with stroke width.